src/Adapter/Order/CommandHandler/AddOrderFromBackOfficeHandler.php

```diff
@@ -123,6 +127,30 @@ public function handle(AddOrderFromBackOfficeCommand $command)
         return new OrderId((int) $paymentModule->currentOrder);
     }
 
+    /**
+     * Saves customer message and link it to the cart.
+     *
+     * @param Cart $cart
+     * @param string $orderMessage
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    private function addOrderMessage(Cart $cart, string $orderMessage): void
+    {
+        $messageId = null;
+        if ($oldMessage = Message::getMessageByCartId((int) $cart->id)) {
+            $messageId = $oldMessage['id_message'];
+        }
+        $message = new Message((int) $messageId);
+        if (Validate::isMessage($orderMessage)) {
```

Comment:
This is the first thing you need to do before continue in this method.
You don't need to waste memory & requests if the message is not valid

```
private function addOrderMessage(Cart $cart, string $orderMessage): void
{
    if (!Validate::isMessage($orderMessage)) {
        throw new OrderContraintException(...);
    }

    //  then you can load the message
}

Reply:
You're right. Fixing it